managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1796,7 +1796,14 @@ void internalMarkDelete(final MarkDeleteEntry mdEntry) {
         // ledger is postponed to when the counter goes to 0.
         PENDING_MARK_DELETED_SUBMITTED_COUNT_UPDATER.incrementAndGet(this);
 
-        lastMarkDeleteEntry = mdEntry;
+        LAST_MARK_DELETE_ENTRY_UPDATER.updateAndGet(this, last -> {
+            if (last != null && last.newPosition.compareTo(mdEntry.newPosition) > 0) {
+                // keep the current value since it's later then the mdEntry.newPosition
```

Comment:
what about adding a DEBUG statement here ?

Reply:
there would be a lot of logging since it's very common that it would happen.

Reply:
then we can add DEBUG in the other branch, the "new" code that you are adding

Reply:
or is this better
```java
                if (log.isDebugEnabled()) {
                    log.debug("Ignoring update {} to lastMarkDeleteEntry {} since current value is later", mdEntry, last);
                }
```